packages/frontend/app/lib/apollo.server.ts

```diff
@@ -16,18 +26,48 @@ BigInt.prototype.toJSON = function (this: bigint) {
   return this.toString()
 }
 
+const authLink = setContext((request, { headers }) => {
+  if (!process.env.SIGNATURE_SECRET || !process.env.SIGNATURE_VERSION)
+    return { headers }
+  const timestamp = Math.round(new Date().getTime() / 1000)
+  const version = process.env.SIGNATURE_VERSION
+
+  const { query, variables, operationName } = request
+  const formattedRequest = {
+    variables,
+    operationName,
+    query: print(query)
+  }
+
+  const payload = `${timestamp}.${canonicalize(formattedRequest)}`
+  const hmac = createHmac('sha256', process.env.SIGNATURE_SECRET)
+  hmac.update(payload)
+  const digest = hmac.digest('hex')
+
+  return {
+    headers: {
+      ...headers,
+      signature: `t=${timestamp}, v${version}=${digest}`
+    }
+  }
+})
+
+const httpLink = createHttpLink({
+  uri: process.env.GRAPHQL_URL ?? 'http://localhost:3001/graphql'
+})
+
```

Comment:
It would be nice if we could share this between `frontend` and `mock-account-servicing-entity`. And potentially `integration` in the future. I recall @sabineschaller brought up a similar need for something else recently (I don't recall what exactly).

I could definitely see a case for a `lib`, `rafiki-lib`, or `internal` (to borrow [a Go convention](https://github.com/golang-standards/project-layout?tab=readme-ov-file#internal)) where this and anything else we wanted to share internally (standardized pino log factory function?).

I guess we dont necessarily need to do that in this PR but but this certainly illustrates the point.